src/plugins/atomic/AtomicWidget.cpp

```diff
@@ -0,0 +1,273 @@
+        auto lines = output.split(QRegularExpression("[\r\n]"),Qt::SkipEmptyParts);
+
+
+        for(const auto& line : lines){
+            qDebug() << line;
+            if(line.contains("status")){
+                parsedLine = QJsonDocument::fromJson(line.toLocal8Bit(), &parseError );
+                if (parsedLine["fields"]["status"].toString().contains("Online")){
+                    bool skip = false;
+                    auto  entry = new OfferEntry(parsedLine["fields"]["price"].toString().split( ' ')[0].toDouble(),parsedLine["fields"]["min_quantity"].toString().split(' ')[0].toDouble(),parsedLine["fields"]["max_quantity"].toString().split(' ')[0].toDouble(), parsedLine["fields"]["address"].toString());
+                    for(const auto& post : *offerList){
+                        if(entry->max == 0 || std::equal(entry->address.begin(), entry->address.end(),post->address.begin(),post->address.end()))
+                            skip = true;
+                    }
+                    if (!skip) {
+                        ui->meta_label->setText("Updated offer book");
+                        offerList->append(QSharedPointer<OfferEntry>(entry));
+                    }
+                }
+            } else if (line.contains("GLIBC_")){
+                QMessageBox::critical(this, "GLIBC outdated", "Upgrade your GLIBC to at least 2.32 to use this tool");
+                clean();
+            }
+        }
+        swap->close();
+        o_model->updateOffers(*offerList);
+        return list;
+    });
+    swap->start(conf()->get(Config::swapPath).toString(), arguments);
+
+
+
+}
+
+AtomicWidget::~AtomicWidget() {
+    qDebug()<< "Exiting widget!!";
+    delete swapDialog;
+    delete o_model;
+    delete offerList;
+    clean();
+    delete procList;
+}
+
+void AtomicWidget::clean() {
+    for (const auto& proc : *procList){
+        proc->kill();
+    }
+    auto cleanWallet =  new QProcess;
+    auto cleanSwap = new QProcess;
+    if(conf()->get(Config::operatingSystem)=="WINDOWS"){
+        (cleanWallet)->start("tskill", QStringList{"monero-wallet-rpc"});
+        (cleanWallet)->start("tskill", QStringList{"swap"});
+    }else {
+        if (constants::networkType==NetworkType::STAGENET){
+            (cleanWallet)->start("pkill", QStringList{"-f", Config::defaultConfigDir().absolutePath() +"/testnet/monero/monero-wallet-rpc"});
+            (cleanSwap)->start("pkill", QStringList{"-f", Config::defaultConfigDir().absolutePath() +
+                                                          "/*"});
+        } else {
+            (cleanWallet)->start("pkill", QStringList{"-f", Config::defaultConfigDir().absolutePath() +
+                                                             "/mainnet/monero/monero-wallet-rpc"});
+            (cleanSwap)->start("pkill", QStringList{"-f", Config::defaultConfigDir().absolutePath() +
+                                                             "/*"});
```

Comment:
The work around would be to either use tasklist/ps to find the pid then kill will QProcess or to start the `monero-wallet-rpc` in feather-atomic detached and supply the port to the swap binary. I will look into this as it may be a better implementation.

Reply:
Unfortunately the swap tool needs to manage it's own `monero-wallet-rpc` instance to function. This is being resolved upstream through UnstoppableSwap/core#114 which will allow this segment of code to be removed.